### PR TITLE
Fix/get default value type

### PR DESCRIPTION
**Pull request by ycherkes**
Fix returned type for the GetDefaultValue function to get `object.Equals(GetDefaultValue(typeof(char)), '\0')` working.

### CODE REVIEW

File: YAXLib/ReflectionUtils.cs [hunk 1]
```diff
@@ -622,24 +617,40 @@ public static bool IsNullable(Type type, out Type? valueType)
...
+                return 0F;
             case TypeCode.Double:
-                return 0.0;
+                return 0D;
             case TypeCode.Decimal:
-                return 0m;
+                return 0M;
             case TypeCode.DateTime:
                 return DateTime.MinValue;
+            case TypeCode.DBNull:
```
Comment (axunonb) [inline YAXLib/ReflectionUtils.cs:645]: Does this make sense to you?

File: YAXLibTests/ReflectionUtilsTest.cs [hunk 2]
```diff
@@ -199,4 +199,25 @@ public void GetSetPropertyValues()
...
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(uint)), Is.EqualTo(0U));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(long)), Is.EqualTo(0L));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(ulong)), Is.EqualTo(0UL));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(float)), Is.EqualTo(0F));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(double)), Is.EqualTo(0D));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(decimal)), Is.EqualTo(0M));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(DateTime)), Is.EqualTo(DateTime.MinValue));
+        Assert.That(ReflectionUtils.GetDefaultValue(typeof(DBNull)), Is.EqualTo(DBNull.Value));
+    }
+}
```
Comment (axunonb) [inline YAXLibTests/ReflectionUtilsTest.cs:223]: Important enough to nail this down

Comment (ycherkes) [reply YAXLibTests/ReflectionUtilsTest.cs:223]: Updated test to show that we compare with default values.

**Comment by axunonb** [comment]
Yep, better have descriptive tests, thanks

**Pull request merged**